### ISSUE TITLE
Warnings upon install

### ISSUE THREAD

**Issue by uvtc** [clojure-emacs/inf-clojure#75]
Installed inf-clojure and received some warnings upon install. This is on Debian Testing, Emacs 24.5.1, lein 2.7.1 on OpenJDK 1.8. Looks like I'm using clojure-mode version 3.1.0-snapshot (20141101.1?).

Here are the warnings:

~~~

Compiling file /home/john/.emacs.d/elpa/inf-clojure-20170507.233/inf-clojure-pkg.el at Mon May  8 14:05:45 2017
Entering directory `/home/john/.emacs.d/elpa/inf-clojure-20170507.233/'

Compiling file /home/john/.emacs.d/elpa/inf-clojure-20170507.233/inf-clojure.el at Mon May  8 14:05:45 2017

In inf-clojure-results-from-process:
inf-clojure.el:943:79:Warning: Use `with-current-buffer' rather than
    save-excursion+set-buffer

In end of data:
inf-clojure.el:1275:1:Warning: the function `clojure-mode-variables' is not
    known to be defined.
~~~

**Comment by uvtc** [comment]
Installed inf-clojure via `M-x package-refresh-contents` and `M-x package-install [RET] inf-clojure`.

**Comment by uvtc** [comment]
Sorry, using the melpa repo.

**Comment by arichiardi** [comment]
I checked and there definitely is a function like that: https://github.com/clojure-emacs/clojure-mode/blob/2d67af4230657fb06f0df4453b9d21031b32c1db/clojure-mode.el#L484

Maybe some require order issue? I am not very familiar about this load order in elisp.

**Comment by uvtc** [comment]
Thank you. Not sure why I had such an old clojure-mode. I've used `M-x package-list`, and interactively installed the latest clojure-mode (which got me 5.7.0-snapshot), and it now has that function (`clojure-mode-variables`) --- the older clojure-mode didn't. (I then had two clojure-mode directories under ~/.emacs.d/elpa and so removed the older one (and then restarted Emacs).)

Looks like I also have clojure-mode-extra-font-locking installed.

Just to test, I removed inf-clojure (just rm -rf'd its directory in ~/.emacs.d/elpa), and reinstalled. It then only gave me that first warning:

~~~
Compiling file /home/john/.emacs.d/elpa/inf-clojure-20170508.2017/inf-clojure-pkg.el at Tue May  9 12:29:38 2017
Entering directory `/home/john/.emacs.d/elpa/inf-clojure-20170508.2017/'

Compiling file /home/john/.emacs.d/elpa/inf-clojure-20170508.2017/inf-clojure.el at Tue May  9 12:29:38 2017

In inf-clojure-results-from-process:
inf-clojure.el:943:79:Warning: Use `with-current-buffer' rather than
    save-excursion+set-buffer
~~~
